syntax = "proto3";
package imakara_admin_common;

service StreamerInfo {
  rpc GetStreamerInfo (StreamerInfoRequest) returns (StreamerInfoResponse);
  rpc ListStreamerInfo (ListStreamerInfoRequest) returns (ListStreamerInfoResponse);
  rpc PutStreamerInfo (PutStreamerInfoRequest) returns (PutStreamerInfoResponse);
}

message ListStreamerInfoRequest {}

message ListStreamerInfoResponse {
  repeated string ids = 1;
}

message PutStreamerInfoRequest {
  string title = 1;
  string description = 2;
  string avatar_url = 3;
  string primary_channel = 4;
}

message PutStreamerInfoResponse {
  ValidationResult result = 1;
  string reason = 2;
}

message StreamerInfoRequest {
  string url = 1;
}

message StreamerInfoResponse {
  ValidationResponse validation_response = 1;
  string title = 2;
  string avatar_url = 3;
  string primary_channel = 4;
}

message ValidationResponse {
  ValidationResult result = 1;
  string reason = 2;
}

enum ValidationResult {
  UNKNOWN = 0;
  SUCCESS = 1;
  FAILURE = 2;
}
